### DISCUSSION TITLE
Select with w-full extends beyond parent width

### DISCUSSION THREAD

**Discussion by Mashiane** [saadeghi/daisyui#1715] [Q&A] [answered]
Hi

How can I fix the width of the select? I have set the width of the select to be w-full, which is the same as the controls on this page.

![image](https://user-images.githubusercontent.com/5799306/228016499-297bcb30-7335-4459-9295-f1514d09be98.png)

It seems the width taken depends on the longest content in the select, whilst other input components fit well inside the parent div.

![image](https://user-images.githubusercontent.com/5799306/228017139-f6b611f5-2e21-4ac3-aac9-974a2351d278.png)

**Comment by saadeghi** [comment, marked as answer]
If `input` or `select` has `w-full` class it will fit 100% of container width.  if there is a `max-w-*` class, you might want to remove it because it limits the max width.   
Here's an example: https://play.tailwindcss.com/0AxEijJiUT

Let me know if you have any questions
